ios/NativeModules/TextileNode.m

```diff
@@ -85,25 +85,31 @@ - (dispatch_queue_t)methodQueue {
       PHFetchResult *allPhotosResult = [PHAsset fetchAssetsWithMediaType:PHAssetMediaTypeImage options:onlyImagesOptions];
       [allPhotosResult enumerateObjectsUsingBlock:^(PHAsset *asset, NSUInteger idx, BOOL *stop) {
 
-        // Get our path right away while we are in Obj-c
-        [asset requestContentEditingInputWithOptions:nil completionHandler:^(PHContentEditingInput * _Nullable contentEditingInput, NSDictionary * _Nonnull info) {
-
-          // get rid of file://
-          NSString *uri = contentEditingInput.fullSizeImageURL.absoluteString;
-          NSString *path = [uri substringFromIndex:7];
-          NSNumber *orientation = contentEditingInput.fullSizeImageOrientation ? [NSNumber numberWithInt:contentEditingInput.fullSizeImageOrientation] : [NSNumber numberWithInt:1];
-          NSFileManager *fileManager = [NSFileManager defaultManager];
-          BOOL isExist = [fileManager fileExistsAtPath:path];
-          if (isExist) {
+        // Get the image data for copying
+        [[PHImageManager defaultManager] requestImageDataForAsset:asset options:nil resultHandler:^(NSData * _Nullable imageData, NSString * _Nullable dataUTI, UIImageOrientation imageOrientation, NSDictionary * _Nullable info) {
+
+          if (imageData) {
+            // Get the original filename to keep it straight on the system.
+            // Alternatively could probably just use timestamp, but this doesn't seem to have any lag
+            NSArray *resources = [PHAssetResource assetResourcesForAsset:asset];
+            NSString *orgFilename = ((PHAssetResource*)resources[0]).originalFilename;
+
+            // Get our path in the tmp directory
+            NSString *path = [[NSTemporaryDirectory()stringByStandardizingPath] stringByAppendingPathComponent:orgFilename];
```

Comment:
Nice. One thought/thing to keep an eye on... Depending on how long we need this image to exist for, we may want to consider writing the image to the documents dir. I'd imagine tmp has some sort of time where the OS deletes the file. Could be fine, though.